tests/pytorch/nightly/roberta-pre.libsonnet

```diff
@@ -19,67 +19,68 @@ local tpus = import 'templates/tpus.libsonnet';
+        '--arch=roberta_base',
+        '--sample-break-mode=complete',
+        '--tokens-per-sample=512',
+        '--optimizer=adam',
+        '--adam-betas',
+        '(0.9,0.98)',
+        '--adam-eps=1e-6',
+        '--clip-norm=0.0',
+        '--lr-scheduler=polynomial_decay',
+        '--lr=0.0005',
+        '--warmup-updates=10000',
+        '--dropout=0.1',
+        '--attention-dropout=0.1',
+        '--weight-decay=0.01',
+        '--update-freq=16',
+        '--log-format=simple',
+        '--train-subset=train',
+        '--valid-subset=valid',
+        '--num_cores=8',
+        '--metrics_debug',
+        '--save-dir=checkpoints',
+        '--log_steps=30',
+        '--skip-invalid-size-inputs-valid-test',
+        '--suppress_loss_report',
+        '--max-epoch=%d' % self.maxEpoch,
+        '--input_shapes',
+      ] + config.paramsOverride.inputShape,
+
     },
     command: utils.scriptCommand(
       |||
         pip install --editable tpu-examples/deps/fairseq
-        python3 \
-          /tpu-examples/deps/fairseq/train.py \
-          /datasets/wikitext-103 \
-          --task=masked_lm --criterion=masked_lm \
-          --arch=roberta_base --sample-break-mode=complete \
-          --tokens-per-sample=512 \
-          --optimizer=adam \
-          --adam-betas='(0.9,0.98)' \
-          --adam-eps=1e-6 \
-          --clip-norm=0.0 \
-          --lr-scheduler=polynomial_decay \
-          --lr=0.0005 \
-          --warmup-updates=10000 \
-          --dropout=0.1 \
-          --attention-dropout=0.1 \
-          --weight-decay=0.01 \
-          --update-freq=16 \
-          --log-format=simple \
-          --train-subset=train \
-          --valid-subset=valid \
-          --num_cores=8 \
-          --metrics_debug \
-          --save-dir=checkpoints \
-          --log_steps=30 \
-          --skip-invalid-size-inputs-valid-test \
-          --suppress_loss_report \
-          --input_shapes 16x512 18x480 21x384 \
-          --max-epoch=%(maxEpoch)d \
-          2>&1 | tee training_logs.txt
+        %(command_string)s 2>&1 | tee training_logs.txt
         wps=$(cat training_logs.txt | grep '| wps ' | tail -1 | grep -o -E ' wps [0-9]+' | sed 's/[^0-9]*//g')
         echo 'final words per second (wps) is' $wps
         test $wps -gt %(wpsTarget)d
-      ||| % self.paramsOverride,
+      ||| % {
+        command_string: utils.toCommandString(config.paramsOverride.trainCommand),
+        wpsTarget: config.paramsOverride.wpsTarget,
+      }
     ),
     volumeMap+: {
       datasets: common.datasetsVolume,
     },
-    podTemplate+:: {
-      spec+: {
-        containerMap+: {
-          train+: {
-            resources+: {
-              requests: {
-                cpu: '9.0',
-                memory: '30Gi',
-                'ephemeral-storage': '10Gi',
-              },
-            },
-          },
-        },
-      },
-    },
+    cpu: '9.0',
+    memory: '30Gi',
+    'ephemeral-storage': '10Gi',
```

Comment:
The base class doesn't use this field like it does `cpu` and `memory`, so this line has no effect. I don't object to removing the ephemeral storage requirement and seeing what happens though.

Reply:
yes - that's correct - the git diff showed the ephemeral-storage setting removed even if the source code here had it - I have removed it from the source code now.